plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,645 @@
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + property);
+                }
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.build();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
+    private List<String> extractColumnNamesFromOrderingScheme(String row)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        if (isNullOrEmpty(row)) {
+            return builder.build();
+        }
+        for (String key : SPLITTER.split(row)) {
+            int left = key.indexOf("\"") + 1;
+            int right = key.lastIndexOf("\"");
+            builder.add(key.substring(left, right).toLowerCase(ENGLISH));
+        }
+        return builder.build();
+    }
+
+    private static SliceWriteFunction uuidWriteFunction()
+    {
+        return (statement, index, value) -> {
+            long high = reverseBytes(value.getLong(0));
+            long low = reverseBytes(value.getLong(SIZE_OF_LONG));
+            UUID uuid = new UUID(high, low);
+            statement.setObject(index, uuid, Types.OTHER);
+        };
+    }
+
+    private static Slice uuidSlice(UUID uuid)
+    {
+        return wrappedLongArray(
+                reverseBytes(uuid.getMostSignificantBits()),
+                reverseBytes(uuid.getLeastSignificantBits()));
+    }
+
+    private ColumnMapping uuidColumnMapping()
+    {
+        return ColumnMapping.sliceMapping(
+                uuidType,
+                (resultSet, columnIndex) -> uuidSlice((UUID) resultSet.getObject(columnIndex)),
+                uuidWriteFunction());
+    }
+
+    public static SliceWriteFunction longDecimalWriteFunction(int precision, int scale)
+    {
+        String bindExpression = format("CAST(? AS decimal(%s, %s))", precision, scale);
+        return new SliceWriteFunction()
+        {
+            @Override
+            public String getBindExpression()
+            {
+                return bindExpression;
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, Slice value)
+                    throws SQLException
+            {
+                statement.setBigDecimal(index, new BigDecimal(value.getLong(0)));
+            }
+        };
+    }
+
+    @Override
+    protected void copyTableSchema(
+            Connection connection,
+            String catalogName,
+            String schemaName,
+            String tableName,
+            String newTableName,
+            List<String> columnNames,
+            List<JdbcColumnHandle> columns,
+            JdbcTableHandle tableHandle)
+    {
+        ImmutableList.Builder<String> columnList = ImmutableList.builder();
```

Comment:
I assume the override is needed because Ignite requires a PRIMARY KEY defined but we don't copy table properties during `CREATE TABLE AS SELECT`?

Reply:
Yes. Ignite table must need a column to be primary key, but it does not allow all the table columns are the primary key, we define the dummy column here is for allow we create a temporary table even the column are allow the primary key(actually it might happen, like `INSERT into ... SELECT (primary keys) FROM`)
Some table properties in Ignite are not allow duplicate, like `cache group` and `cache name`, those two properties indicate a unique cache thus it not allow same properties occurs in Ignite. 
Another properties are meaning less when we create a temporary table, like backups, sync mode.

My opinion is that we could only need care about the primary key here.

Reply:
Please add a comment in code with information about why the override is required.